Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx

```diff
@@ -338,8 +345,303 @@ float TOFChannelData::integral(int ch) const
+        float intmax = fitValues[1] + 5 * fitValues[2]; // mean + 5*sigma
+
+        if (intmin < -mRange) {
+          intmin = -mRange;
+        }
+        if (intmax < -mRange) {
+          intmax = -mRange;
+        }
+        if (intmin > mRange) {
+          intmin = mRange;
+        }
+        if (intmax > mRange) {
+          intmax = mRange;
+        }
+
+        xp[goodpoints] = ipair + 0.5;                                  // pair index
+        exp[goodpoints] = 0.0;                                         // error on pair index (dummy since it is on the pair index)
+        deltat[goodpoints] = fitValues[1];                             // delta between offsets from channels in pair (from the fit) - in ps
+        edeltat[goodpoints] = 20 + fitValues[2] / sqrt(entriesInPair); // TODO: for now put by default to 20 ps since it was seen to be reasonable; but it should come from the fit: who gives us the error from the fit ??????
+        localFitter.AddPoint(&(xp[goodpoints]), deltat[goodpoints], edeltat[goodpoints]);
+        goodpoints++;
+        int ch1 = ipair % 96;
+        int ch2 = ipair / 96 ? ch1 + 48 : ch1 + 1;
+        isChON[ch1] = true;
+        isChON[ch2] = true;
+
+        float fractionUnderPeak = entriesInPair > 0 ? c->integral(ich, intmin, intmax) / entriesInPair : 0;
+        // we keep as fractionUnderPeak of the channel the largest one that is found in the 3 possible pairs with that channel (for both channels ch1 and ch2 in the pair)
+        if (fracUnderPeak[ch1] < fractionUnderPeak) {
+          fracUnderPeak[ch1] = fractionUnderPeak;
+        }
+        if (fracUnderPeak[ch2] < fractionUnderPeak) {
+          fracUnderPeak[ch2] = fractionUnderPeak;
+        }
+      } // end loop pairs
+
+      // fit strip offset
+      if (goodpoints == 0) {
+        continue;
+      }
+      LOG(DEBUG) << "We found " << goodpoints << " good points for strip " << istrip << " in sector " << sector << " --> we can fit the TGraph";
+
+      bool isFirst = true;
+      int nparams = 0;
+      LOG(DEBUG) << "N parameters before fixing = " << localFitter.GetNumberFreeParameters();
+
+      // we fix to zero the parameters that have no entry, plus the first one that we find, which we will use as reference for the other offsets
+      for (int i = 0; i < 96; ++i) {
+        if (isChON[i]) {
+          if (isFirst) {
+            localFitter.FixParameter(i, 0.);
+            isFirst = false;
+          } else {
+            nparams++;
+          }
+        } else {
+          localFitter.FixParameter(i, 0.);
+        }
+      }
+
+      LOG(DEBUG) << "Strip = " << istrip << " fitted by thread = " << ithread << ", goodpoints = " << goodpoints << ", number of free parameters = "
+                 << localFitter.GetNumberFreeParameters() << ",  NDF = " << goodpoints - localFitter.GetNumberFreeParameters();
+
+      if (goodpoints <= localFitter.GetNumberFreeParameters()) {
```

Comment:
Ciao @noferini , 

Note that here I replaced: `if (goodpoints < localFitter.GetNumberFreeParameters())` with `if (goodpoints <= localFitter.GetNumberFreeParameters())` (note the "<=").